tests/unit/vault/test_shares.py

```diff
@@ -498,3 +498,90 @@ def test__mint_shares_with_zero_total_supply_positive_assets(
     # shares should be minted at 1:1
     assert vault.balanceOf(fish) == amount
     assert vault.pricePerShare() > (10 ** vault.decimals())
+
+
+def test__mint_shares_with_zero_total_supply_positive_assets(
+    asset, fish_amount, fish, initial_set_up, gov
+):
+    amount = fish_amount // 10
+    first_profit = fish_amount // 10
+
+    vault, strategy, _ = initial_set_up(asset, gov, amount, fish)
+    create_profit(asset, strategy, gov, vault, first_profit)
+    vault.update_debt(strategy, int(0), sender=gov)
+    assert (
+        vault.totalSupply() > amount
+    )  # there are more shares than deposits (due to profit unlock)
+
+    # User redeems shares
+    vault.redeem(vault.balanceOf(fish), fish, fish, sender=fish)
+
+    assert vault.totalSupply() > 0
+
+    ape.chain.mine(timestamp=ape.chain.pending_timestamp + 14 * 24 * 3600)
+
+    assert vault.totalSupply() == 0
+
+    vault.mint(amount, fish, sender=fish)
+
+    # shares should be minted at 1:1
+    assert vault.balanceOf(fish) == amount
+    assert vault.pricePerShare() > (10 ** vault.decimals())
+
+
+def test__deposit_with_zero_total_assets_positive_supply(
+    asset, fish_amount, fish, initial_set_up, gov
+):
+    amount = fish_amount // 10
+
+    vault, strategy, _ = initial_set_up(asset, gov, amount, fish)
+
+    # Create a loss
+    asset.transfer(gov, amount, sender=strategy)
+    strategy.report(sender=gov)
+
+    assert strategy.convertToAssets(amount) == 0
+
+    vault.process_report(strategy, sender=gov)
+
+    assert vault.totalAssets() == 0
+    assert vault.totalSupply() != 0
+
+    with ape.reverts("cannot mint zero"):
+        vault.deposit(amount, fish, sender=fish)
+
+    # shares should not be
+    assert vault.balanceOf(fish) == amount
+    assert vault.pricePerShare() == 0
+    assert vault.convertToShares(amount) == 0
+    assert vault.convertToAssets(amount) == 0
+    # assert vault.maxDeposit(fish) == 0
```

Comment:
`maxDeposit()` won't return 0 but `uint256.max`? https://github.com/yearn/yearn-vaults-v3/blob/60524bd4d80419672429ab7218d95e7b6e353375/contracts/VaultV3.vy#L529

Reply:
yes, maxDeposit and maxMint wont be accurate currently in this scenario

Not sure any good way to fix it though without adding way to much complexity and extra gas.

Seems like such a minor edge case that it is not worth it.

Reply:
Realistically if this happens no-one should be trying to deposit and a manager should shutdown the vault anyway which will reduce the deposit limit.